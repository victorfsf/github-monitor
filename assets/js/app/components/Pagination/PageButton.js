import React from 'react';
import PropTypes from 'prop-types';


const PageButton = props => (
  <li className={`page-item ${props.disabled ? 'disabled' : ''}`}>
    <button
      type="button"
      className="page-link"
      onClick={props.onClick}
    >
      {props.children}
    </button>
  </li>
);


PageButton.propTypes = {
  disabled: PropTypes.bool,
  children: PropTypes.node,
  onClick: PropTypes.func.isRequired,
};


PageButton.defaultProps = {
  disabled: false,
  children: null,
};


export default PageButton;
