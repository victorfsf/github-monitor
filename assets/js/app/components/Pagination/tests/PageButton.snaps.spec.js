import React from 'react';
import renderer from 'react-test-renderer';
import PageButton from 'app/components/Pagination/PageButton';


describe('PageButton', () => {
  let Component;
  let tree;

  test('does not render without onClick', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<PageButton disabled>Child</PageButton>);
    expect(global.console.error).toHaveBeenCalledTimes(1);
  });

  test('renders', () => {
    Component = renderer.create(
      <PageButton onClick={e => e}>
        Child
      </PageButton>,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders disabled', () => {
    Component = renderer.create(
      <PageButton disabled onClick={e => e}>
        Child
      </PageButton>,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
