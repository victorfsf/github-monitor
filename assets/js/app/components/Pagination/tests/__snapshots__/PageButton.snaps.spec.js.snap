// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButton renders 1`] = `
<li
  className="page-item "
>
  <button
    className="page-link"
    onClick={[Function]}
    type="button"
  >
    Child
  </button>
</li>
`;

exports[`PageButton renders disabled 1`] = `
<li
  className="page-item disabled"
>
  <button
    className="page-link"
    onClick={[Function]}
    type="button"
  >
    Child
  </button>
</li>
`;
